zerver/actions/message_delete.py

```diff
@@ -146,3 +147,33 @@ def do_delete_messages_by_sender(user: UserProfile) -> None:
     )
     if message_ids:
         move_messages_to_archive(message_ids, chunk_size=retention.STREAM_MESSAGE_BATCH_SIZE)
+
+
+def delete_deactivated_user_messages(
+    realm: Realm, user_profile: UserProfile, message_delete_action: list[int]
+) -> None:
+    if Message.NO_DELETE_ACTION in message_delete_action:
+        return
+    user_stream_messages = list(
+        Message.objects.filter(
+            realm_id=user_profile.realm_id,
+            sender=user_profile,
+            recipient__type=Recipient.STREAM,
+        ).only("id", "recipient__type", "recipient__type_id")
+    )
+    private_stream_messages, public_stream_messages = classify_stream_messages(
+        realm, user_stream_messages
+    )
+    if Message.DELETE_PUBLIC_STREAM_MESSAGES in message_delete_action:
+        do_delete_messages(user_profile.realm, public_stream_messages, acting_user=user_profile)
+    if Message.DELETE_PRIVATE_STREAM_MESSAGES in message_delete_action:
+        do_delete_messages(user_profile.realm, private_stream_messages, acting_user=user_profile)
+    if Message.DELETE_DIRECT_MESSAGES in message_delete_action:
+        user_direct_messages = list(
+            Message.objects.filter(
+                realm_id=user_profile.realm_id,
+                sender=user_profile,
+                recipient__type__in=[Recipient.PERSONAL, Recipient.DIRECT_MESSAGE_GROUP],
+            )
+        )
+        do_delete_messages(user_profile.realm, user_direct_messages, acting_user=user_profile)
```

Comment:
Comment in `do_delete_messages` mentions that caller will need to handle 1:1 DMs. And reading the code it seems that you would need to handled 1:1 DM with each user separately, otherwise users other than the sender and receiver will get the events. You should check this once if that is the case and then you would need to update that accordingly.